charts/cloudzero-agent/docs/cert-trouble-shooting.md

```diff
@@ -0,0 +1,367 @@
+2. [Common Certificate Issues](#common-certificate-issues)
+3. [Diagnostic Steps](#diagnostic-steps)
+    - [1. Verify Certificate Configuration](#1-verify-certificate-configuration)
+    - [2. Check Certificate Status](#2-check-certificate-status)
+    - [3. Inspect Admission Webhook Configuration](#3-inspect-admission-webhook-configuration)
+    - [4. Review Pod Logs](#4-review-pod-logs)
+    - [5. Validate Secret Contents](#5-validate-secret-contents)
+4. [Resolution Strategies](#resolution-strategies)
+    - [1. Reinstall or Update `cert-manager`](#1-reinstall-or-update-cert-manager)
+    - [2. Regenerate Certificates Using `cloudzero-certificate` Chart](#2-regenerate-certificates-using-cloudzero-certificate-chart)
+    - [3. Use a Custom Certificate](#3-use-a-custom-certificate)
+    - [4. Correct Certificate References in Configuration](#4-correct-certificate-references-in-configuration)
+5. [Best Practices](#best-practices)
+6. [Additional Resources](#additional-resources)
+
+---
+
+## Understanding Certificate Management
+
+The CloudZero Agent Helm chart deploys a `ValidatingWebhookConfiguration` resource that requires TLS certificates to secure communication with the Kubernetes API server. By default, the chart uses [`cert-manager`](https://cert-manager.io/) to automate the creation and management of these certificates. However, the chart also supports alternative methods:
+
+1. **Using `cert-manager`**: Automates certificate issuance and renewal.
+2. **Using `cloudzero-certificate` Chart**: Manually manages certificates via a separate Helm chart.
+3. **Bringing Your Own Certificate**: Utilizes externally managed certificates provided by an existing certificate manager.
+
+Proper configuration and management of these certificates are essential for the admission controller to function correctly.
+
+---
+
+## Common Certificate Issues
+
+1. **Certificate Not Issued**: The TLS certificate required for the webhook is not created.
+2. **Certificate Expired**: The existing certificate has expired, causing webhook validation failures.
+3. **Invalid Certificate Configuration**: Incorrect references to certificate secrets or misconfigured issuer settings.
+4. **Webhook Fails to Authenticate**: Mismatch between the CA bundle in the webhook configuration and the actual certificate.
+5. **`cert-manager` CRDs Not Installed**: Missing Custom Resource Definitions required by `cert-manager`.
+
+---
+
+## Diagnostic Steps
+
+### 1. Verify Certificate Configuration
+
+Ensure that the certificate management configuration in your `values.yaml` or `configuration.example.yaml` is correctly set based on your chosen method (`cert-manager`, `cloudzero-certificate`, or custom certificates).
+
+- **Using `cert-manager`**:
+  ```yaml
+  cert-manager:
+    enabled: true
+  insightsController:
+    cert-manager:
+      enabled: true
+  ```
+  
+- **Using `cloudzero-certificate` Chart**:
```

Comment:
This is the same configuration shown for using custom certificates -- is that OK?